### PR TITLE
Significantly reduce volume of disposals machinery when processing a ton of items

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Adds a short cooldown on the recycler when it consumes items, as well as disposals machinery processing items in general, preventing them from playing a ton of overlapping sounds at once.

It also fixes some checks that were already there but inverted (read: broken).

## Why It's Good For The Game
When stuff has piled up in cargo disposals and someone finally pulls the lever, any poor sap close to the recycler will have their ears destroyed. This isn't really great, [as funny as it can be](https://cdn.discordapp.com/attachments/145700319819464704/1129853880008724550/2023-07-15_21-15-27.mp4) (VOLUME WARNING)

## Testing
Spawned 100 pieces of paper, put them through disposals, my ears felt fine afterwards.

## Changelog
:cl:
tweak: Recyclers and disposals are now a lot quieter when processing a ton of items at once.
/:cl:

### CODE REVIEW

**Comment by CinnamonSnowball** [comment]
Could this be applied to disposal bins/chutes as well, I beg thee?

**Comment by lewcc** [comment]
> Could this be applied to disposal bins/chutes as well, I beg thee?

Done, thanks for the heads up! Turns out that's actually where most of the sound was coming from.

File: code/game/machinery/recycler.dm [hunk 1]
```diff
@@ -137,8 +139,9 @@
 			playsound(loc, 'sound/machines/buzz-sigh.ogg', 50, 0)
 			AM.forceMove(loc)
 
-	if(items_recycled && sound)
+	if(items_recycled && sound && (last_consumption_sound + 0.5 SECONDS) < world.time)
```
Comment (Sirryan2002) [inline code/game/machinery/recycler.dm:142]: make these SECONDS into a local define so we're not using magic numbers

File: code/modules/recycling/disposal.dm [hunk 2]
```diff
@@ -464,7 +464,7 @@
 		H.tomail = 1
 
 	sleep(10)
-	if(last_sound < world.time + 1)
+	if(last_sound + 0.1 SECONDS < world.time)
```
Comment (Sirryan2002) [inline code/modules/recycling/disposal.dm:467]: same here and on L502

Comment (lewcc) [reply code/modules/recycling/disposal.dm]: Made this one a global define as we have to use it in two files

File: code/modules/recycling/disposal.dm [hunk 3]
```diff
@@ -499,7 +501,10 @@
 /obj/machinery/disposal/proc/expel(obj/structure/disposalholder/H)
 
 	var/turf/target
-	playsound(src, 'sound/machines/hiss.ogg', 50, 0, 0)
+	if(last_sound + DISPOSAL_SOUND_COOLDOWN < world.time)
+		playsound(src, 'sound/machines/hiss.ogg', 50, 0, 0)
```
Comment (Sirryan2002) [inline code/modules/recycling/disposal.dm:505]:
```suggestion
		playsound(src, 'sound/machines/hiss.ogg', 50, 0, FALSE)
```

**Review by Sirryan2002 — APPROVED** [review]
few 0->FALSE conversions, otherwise approval

**Review by S34NW — APPROVED** [review]

**Pull request merged**